pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -173,62 +178,58 @@ func (r *Reconciler) DetectOperation(ctx context.Context, log logr.Logger, tinke
 	}
 
 	wantVersionChange := currentKCP.Spec.Version != tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Version
-	cpWantScaleChange := *currentKCP.Spec.Replicas != *tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Replicas
-	workerWantScaleChange, err := r.WorkerReplicasDiff(ctx, tinkerbellScope)
-	if err != nil {
-		return "", err
-	}
-	wantScaleChange := workerWantScaleChange || cpWantScaleChange
 
 	// The restriction that not allowing scaling and rolling is covered in webhook.
 	op := NoChange
-	switch {
-	case wantScaleChange:
-		op = ScaleOperation
-	case wantVersionChange:
+	if wantVersionChange {
 		op = K8sVersionUpgradeOperation
 	}
 	log.Info("Operation detected", "operation", op)
```

Comment:
This log message won't be printed for create operation cuz that's returning instead of just modifying the op https://github.com/aws/eks-anywhere/pull/5246/files#diff-912930497d9a5035c7a0b2e57b0740c683c897b56baf1628991e8e43146c5f68R176-R177